ichnaea/data/tests/test_station.py

```diff
@@ -256,143 +256,135 @@ def test_blocklist_skip(self, celery, redis, session):
-                lat=obs.lat - 0.0003, source=source, **self.key(obs)
-            )
-            obs3 = self.obs_factory(
-                lon=obs.lon + 0.0002, source=source, **self.key(obs)
-            )
-            obs4 = self.obs_factory(
-                lon=obs.lon - 0.0004, source=source, **self.key(obs)
-            )
-            self.queue_and_update(celery, [obs, obs1, obs2, obs3, obs4])
-
-            self.check_areas(celery, [obs])
-            station = self.get_station(session, obs)
-            assert round(station.lat, 7) == round(obs.lat - 0.00004, 7)
-            assert round(station.max_lat, 7) == round(obs.lat + 0.0001, 7)
-            assert round(station.min_lat, 7) == round(obs.lat - 0.0003, 7)
-            assert round(station.lon, 7) == round(obs.lon - 0.00004, 7)
-            assert round(station.max_lon, 7) == round(obs.lon + 0.0002, 7)
-            assert round(station.min_lon, 7) == round(obs.lon - 0.0004, 7)
-            assert station.radius == 38
-            assert station.region == "GB"
-            assert station.samples == 5
-            assert station.source == source
-            assert round(station.weight, 2) == 5.0
-            self.check_blocked(station, None)
-            self.check_dates(station, self.today, self.today, self.today)
+    @pytest.mark.parametrize("source", (ReportSource.gnss, ReportSource.query))
+    def test_new(self, celery, session, metricsmock, source):
+        obs = self.obs_factory.build(source=source)
+        obs1 = self.obs_factory(lat=obs.lat + 0.0001, source=source, **self.key(obs))
+        obs2 = self.obs_factory(lat=obs.lat - 0.0003, source=source, **self.key(obs))
+        obs3 = self.obs_factory(lon=obs.lon + 0.0002, source=source, **self.key(obs))
+        obs4 = self.obs_factory(lon=obs.lon - 0.0004, source=source, **self.key(obs))
+        self.queue_and_update(celery, [obs, obs1, obs2, obs3, obs4])
+
+        self.check_areas(celery, [obs])
+        station = self.get_station(session, obs)
+        assert round(station.lat, 7) == round(obs.lat - 0.00004, 7)
+        assert round(station.max_lat, 7) == round(obs.lat + 0.0001, 7)
+        assert round(station.min_lat, 7) == round(obs.lat - 0.0003, 7)
+        assert round(station.lon, 7) == round(obs.lon - 0.00004, 7)
+        assert round(station.max_lon, 7) == round(obs.lon + 0.0002, 7)
+        assert round(station.min_lon, 7) == round(obs.lon - 0.0004, 7)
+        assert station.radius == 38
+        assert station.region == "GB"
+        assert station.samples == 5
+        assert station.source == source
+        assert round(station.weight, 2) == 5.0
+        self.check_blocked(station, None)
+        self.check_dates(station, self.today, self.today, self.today)
 
         assert (
             len(
                 metricsmock.filter_records(
                     "incr", "data.observation.insert", value=5, tags=[self.type_tag]
                 )
             )
-            == 2
+            == 1
```

Comment:
Now that `queue_and_update` is run once per test rather than twice in a loop, this metric, and the following `data.station.new` metric, is only incremented once.